client/web/src/App.jsx

```diff
@@ -168,12 +189,27 @@ const App = () => {
   }
 
   // Use custom hooks
-  const { send, connectSocket, closeSocket } = useWebsocket(handleSocketOnOpen,handleSocketOnMessage);
+  const { socketRef, send, connectSocket, closeSocket } = useWebsocket(token, handleSocketOnOpen,handleSocketOnMessage, selectedModel);
```

Comment:
Another issue is it seems the token is access_token while we expect an id_token https://stackoverflow.com/questions/49565028/python-google-oauth2-wrong-number-of-segments-in-token